transport/http2_client.go

```diff
@@ -861,6 +862,24 @@ func (t *http2Client) operateHeaders(frame *http2.MetaHeadersFrame) {
 	for _, hf := range frame.Fields {
 		state.processHeaderField(hf)
 	}
+	// If grpc status doesn't exist.
+	if !state.statusExists {
+		// Check if http status exists.
+		if !state.hstatusExists {
```

Comment:
Maybe we should not try to read the content of state if stats.err != nil.
For example in the case where http-status is not a valid number and strconv.Atoi() returns an error.